lib/data-types.js

```diff
@@ -460,6 +460,59 @@ class TIME extends ABSTRACT {
   toSql() {
     return 'TIME';
   }
+
+  validate(value) {
+    if (value instanceof Date) {
+      if (Number.isNaN(value.getTime())) {
+        throw new sequelizeErrors.ValidationError(util.format('%j is not a valid time', value));
+      }
+
+      return true;
+    }
+
+    if (typeof value !== 'string' || !/^-?\d+:[0-5][0-9]:[0-5][0-9](?:\.\d+)?$/.test(value)) {
+      throw new sequelizeErrors.ValidationError(util.format('%j is not a valid time', value));
+    }
+
+    return true;
+  }
+
+  _stringify(date) {
+    if (date instanceof Date) {
+      return moment(date).format('HH:mm:ss.SSS');
+    }
+
+    return date;
+  }
+
+  _sanitize(value, options) {
+    if ((!options || !options.raw) && value instanceof Date) {
+      return moment(value).format('HH:mm:ss.SSS');
+    }
+
+    return value;
+  }
+
+  _isChanged(value, originalValue) {
+    if (
```

Comment:
For this one, I think we should first normalize `value` & `originalValue` to strings if they're Date objects. This way `_isChanged(new Date('2022-01-15T12:00:00Z'), '12:00:00.000')` will return `false`

Reply:
So we should convert the date object to `HH:mm:ss.SSS` format string and then compare them right?

Reply:
Yes, I think we can call stringify on the inputs :)
We should also make sure `12:00:00` and `12:00:00.000` are equal